packages/amplify-graphql-rds-model-transformer/rds-lambda/clients/postgres/PostgresPasswordClient.ts

```diff
@@ -0,0 +1,36 @@
+import Knex from 'knex';
+import { PostgresClient } from './PostgresClient.js';
+
+/**
+ * Postgres client with password connection implementation
+ */
+export class PostgresPasswordClient extends PostgresClient {
+    private clientPromise: Promise<any>;
+    constructor() {
+      super();
+      this.clientPromise = this.getClient();
+      this.clientPromise.then((value) => this.client = value);
+    }
+
+    getClient = async (): Promise<any> => {
+      if (this.client) {
+        return this.client;
+      }
+      if (this.clientPromise) {
+        await this.clientPromise;
+        return this.client;
+      }
+      return Knex({
+        client: 'pg',
+        connection: {
+          host: process.env.DATABASE_HOST,
+          port: Number.parseInt(process.env.DATABASE_PORT ?? '5432', 10),
+          user: 'postgres',
+          password: 'postgres',
+          database: 'postgres',
```

Comment:
The database name should also be fetched dynamically either via environment variables or the calling lambda context.

Reply:
Similarly, the username and password will need to be fetched from parameter store inside the lambda. The parameter names for these secrets will need to be passed to the lambda via environment or calling context.

Reply:
Yep, this support is all gonna be dynamic as we start to complete and incorporate those workflows